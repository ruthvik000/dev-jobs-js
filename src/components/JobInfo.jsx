import React from 'react';
import DateFormat from "./DateFormat";

// Component to render Job Info results.
function JobInfo(props) {
    const jobDetails = props.jobInfo;
    if (props.jobInfo) {
        return (
            <div>
                <div className='job-info'>
                    <img className='w3-round-small job-info-image'
                         style={{float: 'left'}}
                         src={jobDetails.company_logo}
                         alt={jobDetails.company}
                         height="100px"
                         width="100px"
                    />
                    <div className='jobsite'>
                        <h2 className='job-details-title'>{jobDetails.company}</h2>
                        <a href={jobDetails.company_url} target="_blank" rel="noopener noreferrer">
                            <button className="company-url-button">Company Site</button>
                        </a>
                    </div>
                </div>
                <div className='job-description-body'>
                    <a href={jobDetails.company_url} target="_blank" rel="noopener noreferrer">
                        <button className="apply-now-button">Apply Now</button>
                    </a>
                    <p className='days-type'>{DateFormat(new Date(jobDetails.created_at))} . {jobDetails.type}</p>
                    <h3 className='job-title'>{jobDetails.title}</h3>
                    <p className='job-location'>{jobDetails.location}</p>
                    <p className='job-description'>{jobDetails.description}</p>
                </div>
                <div className='how-to-apply'>
                    <h5>How to Apply</h5>
                    <p>{jobDetails.how_to_apply}</p>
                </div>
            </div>
        );
    }
}

export default (JobInfo)