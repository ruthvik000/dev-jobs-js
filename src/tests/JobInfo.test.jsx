import React from "react";
import {shallow} from 'enzyme'
import JobInfo from "../components/JobInfo";

const result =
    {
        company_logo: 'xyz',
        company: 'Company 1',
        title: 'Java Developer',
        created_at: '2/29/2021 4:52:48 PM UTC',
        type: 'Full Time',
        location: 'New York',
        description: 'Need an experienced developer with 5 years of real-time experience.',
        how_to_apply: 'Visit Careers Website'
    }

describe('Job Info component', () => {
    it('should match the snapshot with zero results', () => {
        const container = shallow(<JobInfo />);
        expect(container.html()).toMatchSnapshot();
    });

    it('should match the snapshot with results', () => {
        const container = shallow(<JobInfo jobInfo={result} />);
        expect(container.html()).toMatchSnapshot();
    });

    describe('validate classes', () => {
        const container = shallow(<JobInfo jobInfo={result} />);

        it('should have an job info class field', () => {
            expect(container.find('div[className="job-info"]').length).toEqual(1);
        });

        it('should have an job details title field', () => {
            expect(container.find('h2[className="job-details-title"]').length).toEqual(1);
        });

        it('should have an company name url button', () => {
            expect(container.find('button[className="company-url-button"]').length).toEqual(1);
        });

        it('should have an job description body', () => {
            expect(container.find('div[className="job-description-body"]').length).toEqual(1);
        });

        it('should have an days and type field', () => {
            expect(container.find('p[className="days-type"]').length).toEqual(1);
        });

        it('should have an job location field', () => {
            expect(container.find('p[className="job-location"]').length).toEqual(1);
        });

        it('should have an job description field', () => {
            expect(container.find('p[className="job-description"]').length).toEqual(1);
        });

        it('should have an how to apply div', () => {
            expect(container.find('div[className="how-to-apply"]').length).toEqual(1);
        });
    });
});