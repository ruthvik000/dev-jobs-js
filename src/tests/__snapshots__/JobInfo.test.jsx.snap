// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Info component should match the snapshot with results 1`] = `"<div><div class=\\"job-info\\"><img class=\\"w3-round-small job-info-image\\" style=\\"float:left\\" src=\\"xyz\\" alt=\\"Company 1\\" height=\\"100px\\" width=\\"100px\\"/><div class=\\"jobsite\\"><h2 class=\\"job-details-title\\">Company 1</h2><a target=\\"_blank\\" rel=\\"noopener noreferrer\\"><button class=\\"company-url-button\\">Company Site</button></a></div></div><div class=\\"job-description-body\\"><a target=\\"_blank\\" rel=\\"noopener noreferrer\\"><button class=\\"apply-now-button\\">Apply Now</button></a><p class=\\"days-type\\">1 month ago.Full Time</p><h3 class=\\"job-title\\">Java Developer</h3><p class=\\"job-location\\">New York</p><p class=\\"job-description\\">Need an experienced developer with 5 years of real-time experience.</p></div><div class=\\"how-to-apply\\"><h5>How to Apply</h5><p>Visit Careers Website</p></div></div>"`;

exports[`Job Info component should match the snapshot with zero results 1`] = `null`;
